src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -133,9 +139,12 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
             CodeBlock.of("%T", it)
           }
         }
-        val superTypes = types + superinterfaces.map { CodeBlock.of("%T", it) }
+        val superTypes = types + superinterfaces.entries.map { (type, init) ->
+            if (init == null)  CodeBlock.of("%T", type) else CodeBlock.of("%T by $init", type)
```

Comment:
I think there’s a small problem here. This calls `$init.toString()` eagerly but we really want to embed the codeblock using `%L`. Otherwise imports will be wrong if the codeblock contains ’em.

Reply:
ie. `CodeBlock.of(%T by %L, type, init)`

Reply:
I was wondering why the imports didn't work consistently